pkg/api/validate/api.go

```diff
@@ -0,0 +1,56 @@
+package validate
+
+import (
+	"fmt"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+)
+
+// Validate validates a OpenShiftManagedCluster struct
+func (v *APIValidator) Validate(new, old *api.OpenShiftManagedCluster, externalOnly bool) (errs []error) {
+	errs = append(errs, validateContainerService(new, externalOnly)...)
+	errs = append(errs, validateUpdateContainerService(new, old, false)...)
+	// this limits use of RunningUnderTest variable inside our validators
+	// TODO: When removed this should be part of common validators
+	for _, app := range new.Properties.AgentPoolProfiles {
+		errs = append(errs, v.validateVMSize(app)...)
```

Comment:
I really dont like how it will propagate runningUnderTest variable... 
```
	for _, app := range cs.Properties.AgentPoolProfiles {
		errs = append(errs, validateVMSize(app, v.runningUnderTest)...)
	}
```
We force VM size in tests:
```
func isValidMasterAndInfraVMSize(size api.VMSize, runningUnderTest bool) bool {
	if runningUnderTest && size == api.StandardD2sV3 {
		return true
	}

	_, found := validMasterAndInfraVMSizes[size]
	return found
}
```

Should we just ditch it and make sure we create the right VM sizes in the first place?

Reply:
This is still pending what we want to do. I would say remove this and control it via manifest file, where it should be.